src/topology/algebra/module.lean

```diff
@@ -801,3 +806,34 @@ equiv_of_inverse (f₁.prod (f₁.proj_ker_of_right_inverse f₂ h)) (f₂.copro
   (equiv_of_right_inverse f₁ f₂ h).symm y = f₂ y.1 + y.2 := rfl
 
 end continuous_linear_equiv
+
+namespace submodule
+
+variables
+{R : Type*} [ring R]
+{M : Type*} [topological_space M] [add_comm_group M] [module R M]
+{M₂ : Type*} [topological_space M₂] [add_comm_group M₂] [module R M₂]
+
+/-- A submodule `p` is called *complemented* if there exists a continuous projection `M →ₗ[R] p`. -/
+def complemented (p : submodule R M) : Prop := ∃ f : M →L[R] p, ∀ x : p, f x = x
+
+lemma complemented.has_closed_complement {p : submodule R M} [t1_space p] (h : complemented p) :
```

Comment:
A missing lemma around here is the fact that a complemented submodule is closed (to prove it, let `f` be a continuous projection on `p`. Then `p` is the kernel of `id - f`, and therefore closed).